lib/instrumentation/modules/@elastic/elasticsearch.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const { setDbContext } = require('../../elasticsearch-shared')
+
+module.exports = function (elasticsearch, agent, { version, enabled }) {
+  if (!enabled) return elasticsearch
+
+  function generateSpan (activeSpans, meta, params) {
+    const span = agent.startSpan(null, 'db.elasticsearch.request')
+    if (span === null) return null
+
+    const { request } = meta
+    span.name = `Elasticsearch: ${params.method} ${params.path}`
+
+    setDbContext(span, params)
+
+    activeSpans.set(request.id, span)
+
+    return span
+  }
+
+  class ApmClient extends elasticsearch.Client {
+    constructor (opts) {
+      super(opts)
+
+      const activeSpans = new Map()
+      let hasPrepareRequestEvent = false
+
+      this.on('prepare-request', (err, { meta, params }) => {
+        hasPrepareRequestEvent = true
+        if (err) {
+          return agent.captureError(err)
+        }
+
+        generateSpan(activeSpans, meta, params)
+      })
+
+      this.on('request', (err, { meta }) => {
```

Comment:
In the `request` event the `body` and `querystring` values will be already serialized, would it be possible to add the `setDbContext` call here?
In this way, we don't need to serialize the body/querystring twice ;)

Reply:
I'd love the make the query more readable in the case of an array of queries. What I do with the old client is format it as NDJSON so that each query is on its own line. If we use the JSON that you formatted, it will be one long line containing all the queries... not really sure how to best do this 🤔

Reply:
What you mean with "array of queries"?

Reply:
Example:

```js
  const body = [
    {},
    {
      query: {
        query_string: {
          query: 'pants'
        }
      }
    }
  ]

  client.msearch({ body }, function () {...})
```

Reply:
In that case you will get a `ndjson` serialized body.